Sources/NIOSSL/SSLContext.swift

```diff
@@ -211,6 +203,58 @@ private func clientPSKCallback(ssl: OpaquePointer?,
     return UInt32(clientPSK.count)
 }
 
+private func sslContextCallback(ssl: OpaquePointer?, arg: UnsafeMutableRawPointer?) -> Int32 {
+    guard let ssl = ssl else {
+        preconditionFailure("""
+            SSL_CTX_set_cert_cb was executed with an invalid ssl pointer.
+            This should not be possible, please file an issue.
+        """)
+    }
+
+    let parentSwiftContext = NIOSSLContext.lookupFromRawContext(ssl: ssl)
+
+    // This is a safe force unwrap as this callback is only register directly after setting the manager
+    var contextManager = parentSwiftContext.customContextManager!
+    
+    // Begin loading a new context
+    let result = contextManager.loadContext(ssl: ssl)
+
+    switch result {
+    case .none:
+        // If we dont have a result yet then we must suspend
+        return -1
+    case .failure:
+        // If loading a context failed then we must signal as such
+        return SSL_TLSEXT_ERR_NOACK
```

Comment:
This looks wrong. The `SSL_TLSEXT_ERR_*` constants are all positive integers. Returning them here will cause this to look like _success_.

I think we need a failure case test here, as that should have caught this behaviour. Based on code reading, it looks like we need to return 0 to fail here.